site-src/geps/gep-957.md

```diff
@@ -0,0 +1,77 @@
+# GEP-957: Destination Port Matching
+
+* Issue: [#957](https://github.com/kubernetes-sigs/gateway-api/issues/957)
+* Status: Provisional
+
+## TLDR
+
+Support destination port matching.
+
+## Goals
+
+* Support destination port matching based on port number.
+
+## Non-Goals
+
+* Support port matching based on named port.
+
+## Introduction
+
+Port matching is a common service mesh use case where traffic policies/rules
+need to be applied to traffic to certain destination ports. For ingress, while
+the API today already supports attaching a route to a specific listener, it may
+be useful to support attaching routes to listener(s) on a specified port. For
+example, some users may only want to expose their application on ports 80 or 443
+on the Gateway.
+
+## API
+
+The proposal is to add a new field `Port` to `ParentRef`:
+
+```go
+type ParentRef struct {
+  ...
+  // Port is the network port this resource targets. It can be interpreted
+  // differently based on the type of parent resource:
+  //
+  // Gateway: All listeners listening on the specified port that also support
+  // this kind of Route.
+  //
+  // Implementations MAY choose to support other parent resources.
+  // If that is the case, they MUST clearly document how Port is
+  // interpreted.
+  //
+  // When unspecified, this will reference the entire resource.
+  // For the purpose of status, an attachment is considered successful as
+  // long as the parent resource accepts it partially. For example, Gateway
+  // listeners can restrict which Routes can attach to them by Route kind,
+  // namespace, or hostname. If 1 of 2 Gateway listeners accept attachment from
+  // the referencing Route, the Route MUST be considered successfully
+  // attached. If no Gateway listeners accept attachment from this Route, the
+  // Route MUST be considered detached from the Gateway.
+  //
+  // Support: Core
+  //
+  // +optional
+  Port *PortNumber `json:"port,omitempty"`
+  ...
+}
+```
+
+## Alternatives
+### 1. Use SectionName in ParentRef for port matching
+Port matching can be supported if SectionName accepts port numbers in addition
+to listener names. This approach results in a less explicit API. It can also
+complicate the validation on SectionName.
```

Comment:
Maybe add a bit more detail here such as an example of how it would complicate SectionName validation and/or how this would provide a poor user experience.